builder/amazon/common/ssm_driver.go

```diff
@@ -0,0 +1,56 @@
+package common
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"os/exec"
+
+	"github.com/hashicorp/packer/packer"
+)
+
+const SessionManagerPluginName string = "session-manager-plugin"
+
+type SSMDriver struct {
+	Ui packer.Ui
+	// Provided for testing purposes; if not specified it defaults to SessionManagerPluginName
+	PluginName string
+}
+
+// StartSession starts an interactive Systems Manager session with a remote instance via the AWS session-manager-plugin
+func (s *SSMDriver) StartSession(sessionData, region, profile, params, endpoint string) error {
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+
+	args := []string{
+		sessionData,
+		region,
+		"StartSession",
+		profile,
+		params,
+		endpoint,
+	}
+
+	if s.PluginName == "" {
+		s.PluginName = SessionManagerPluginName
+	}
+
+	if _, err := exec.LookPath(s.PluginName); err != nil {
+		return err
+	}
+
+	log.Printf("Attempting to start session with the following args: %v", args)
+	cmd := exec.Command(s.PluginName, args...)
```

Comment:
Here I think we should pass a context into `StartSession`, and then call 
```suggestion
	cmd := exec.CommandContext(ctx, s.PluginName, args...)
```

Later on when that context is cancelled ( for example in `StepCreateSSMTunnel.Cleanup` or if Packer crashed ) the command is killed and the user wont have to `terminate the session manually`.

Reply:
> Hey Super nice !! Love it :).
> I was thinking that Packer should also be responsible for the killing of the proxy since it starts it, also What happens when two ssm builders run at the same time ?

I want to run some tests and screen captures here to confirm. But in my testing I've found that once the session is terminated via the SDK Amazon terminates the session and the call to the session-manage-plugin exits because the upstream agent no longer has an active connection. Thus not having to worry about shutting down the proxy.  

 Can we start multiple proxy at once and they will do their thing on differents ports ?
Yes, this has been tested as long as they have different ports communication works as expected. I have not tested if there is a limit to this. But in thinking about this question a bit I can see a potential race condition with how the code is currently written; more specifially if two parallel builds get the same port number and try to open the tunnel at the same time I think one build will fail.

Reply:
nice :D 🦆